app/src/main/java/com/github/palFinderTeam/palfinder/ui/login/LoginActivity.kt

```diff
@@ -173,20 +224,109 @@ class LoginActivity : AppCompatActivity() {
                     Log.w(TAG, "Google sign in failed", e)
                 }
             }
+            REQUEST_CODE_GIS_SAVE_PASSWORD -> {
+                Log.d(TAG, "in save password result")
+                if (resultCode == Activity.RESULT_OK) {
+                    /* password was saved */
+                    Toast.makeText(
+                        baseContext, "password saved",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                } else if (resultCode == Activity.RESULT_CANCELED) {
+                    /* password saving was cancelled */
+                    Toast.makeText(
+                        baseContext, "password not saved",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+                updateUI(auth.currentUser)
+            }
         }
     }
 
+    private fun createAccount(email: String, password: String) {
+        // [START create_user_with_email]
+        auth.createUserWithEmailAndPassword(email, password)
+            .addOnCompleteListener(this) { task ->
+                if (task.isSuccessful) {
+                    // Sign in success, update UI with the signed-in user's information
+                    Log.d(TAG, "createUserWithEmail:success")
+                    savePassword(email,password)
+                } else {
+                    // If user already in database, sign in
+                    if(task.exception is FirebaseAuthUserCollisionException){
+                        signIn(email, password)
```

Comment:
Since you re perform the checks here it actually sign in even though it thought that the email was not taken, which indeed works.